### PR TITLE
end-user: add terraform doc

### PR DESCRIPTION
**Pull request by jerryB1**

### CODE REVIEW

File: source/end-user/integration/terraform.rst [hunk 1]
```diff
@@ -1,3 +1,156 @@
...
+
+   module "sample" {
+     source = "modules/sample"
+   }
+
+.. note::
+
+   ``provider:`` Before we can use the OpenStack provider, we configured the proper credentials.
+   Download the ``OpenStack RC File v3/v2`` from the horizon dashboard and evaluate the file with 
+   the command ``source``. Enter your OpenStack Password for youre project.
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Also document how to be able to not use the openrc file (https://www.terraform.io/docs/providers/openstack/index.html#example-usage). At least the use of clouds.yml file should be documented as well.

File: source/end-user/integration/terraform.rst [hunk 2]
```diff
@@ -1,3 +1,156 @@
...
+============
+
+* Create a ``variables.tf`` file below the sample directory
+
+.. code-block:: json
+
+   variable "openstack_keypair" {
+       description = "The keypair to be used."
+       default  = "authentication_key"
+   }
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Is it not possible to also create the key pair with Terraform?

File: source/end-user/integration/terraform.rst [hunk 3]
```diff
@@ -1,3 +1,156 @@
...
+
+   variable "openstack_network" {
+       description = "The network to be used."
+       default  = "net-to-DOMAINNAME-public"
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Use "network" as default value.

File: source/end-user/integration/terraform.rst [hunk 4]
```diff
@@ -1,3 +1,156 @@
...
+   }
+
+   variable "openstack_network_floatingip_pool" {
+       description = "The network floatingip pool to be used."
+       default  = "DOMAINNAME-public"
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Use "public" as default value.

File: source/end-user/integration/terraform.rst [hunk 5]
```diff
@@ -1,3 +1,156 @@
...
+   }
+
+   variable "openstack_flavor_id" {
+       description = "The flavor id to be used."
+       default  = "122"
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Is it necessary to use IDs? Is it not possible to use flavor names?

File: source/end-user/integration/terraform.rst [hunk 6]
```diff
@@ -1,3 +1,156 @@
...
+       description = "The image name to be used."
+       default  = "Ubuntu 18.04 (Bionic Beaver)"
+   }
+
+
+.. note::
+
+The variables must be adjusted according to the project used.
+
+* Start the deployment now
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Describe how to run the deployment

File: source/end-user/integration/terraform.rst [hunk 7]
```diff
@@ -1,3 +1,156 @@
...
+
+Terraform commands
+==================
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: This section is not necessary.

File: source/end-user/integration/terraform.rst [hunk 8]
```diff
@@ -1,3 +1,156 @@
...
+* Create a ``modules.tf`` file
+
+.. code-block:: json
+
+   provider "openstack" {
+     version = "~> 1.6"
+   }
+
+   terraform {
+     required_version = "= 0.11.7"
```
Comment (berendt) [inline source/end-user/integration/terraform.rst]: Why do we have to pin the required version?

**Review by berendt — CHANGES_REQUESTED** [review]

File: source/end-user/integration/terraform.rst (hunk 2 above)
Comment (jerryB1) [reply source/end-user/integration/terraform.rst]: Yes is possible

File: source/end-user/integration/terraform.rst (hunk 5 above)
Comment (jerryB1) [reply source/end-user/integration/terraform.rst]: i added flavor name

File: source/end-user/integration/terraform.rst (hunk 8 above)
Comment (jerryB1) [reply source/end-user/integration/terraform.rst]: No we need not pin the required version.

**Pull request merged**